spec/categorical_spec.rb

```diff
@@ -0,0 +1,1725 @@
+      end
+      
+      context "manual base category" do
+        let(:dv) { Daru::Vector.new [:a, 1, :a, 1, :c], type: :category, name: :abc }
+        subject { dv.contrast_code }
+        before do
+          dv.coding_scheme = :simple
+          dv.base_category = :c
+        end
+        
+        it { is_expected.to be_a Daru::DataFrame }
+        its(:shape) { is_expected.to eq [5, 2] }
+        its(:'abc_a.to_a') { is_expected.to eq [2/3.0, -1/3.0, 2/3.0, -1/3.0, -1/3.0] }
+        its(:'abc_1.to_a') { is_expected.to eq [-1/3.0, 2/3.0, -1/3.0, 2/3.0, -1/3.0] }
+      end
+    end
+
+    context "helmert coding" do
+      let(:dv) { Daru::Vector.new [:a, 1, :a, 1, :c], type: :category, name: :abc }
+      subject { dv.contrast_code }
+      before { dv.coding_scheme = :helmert }
+
+      it { is_expected.to be_a Daru::DataFrame }
+      its(:shape) { is_expected.to eq [5, 2] }
+      its(:'abc_a.to_a') { is_expected.to eq [2/3.0, -1/3.0, 2/3.0, -1/3.0, -1/3.0] }
+      its(:'abc_1.to_a') { is_expected.to eq [0, 1/2.0, 0, 1/2.0, -1/2.0] }
+    end
+
+    context "deviation coding" do
+      let(:dv) { Daru::Vector.new [:a, 1, :a, 1, :c], type: :category, name: :abc }
+      subject { dv.contrast_code }
+      before { dv.coding_scheme = :deviation }
+
+      it { is_expected.to be_a Daru::DataFrame }
+      its(:shape) { is_expected.to eq [5, 2] }
+      its(:'abc_a.to_a') { is_expected.to eq [1, 0, 1, 0, -1] }
+      its(:'abc_1.to_a') { is_expected.to eq [0, 1, 0, 1, -1] }
+    end
+
+    context "user-defined coding" do
+      let(:df) do
+        Daru::DataFrame.new({
+          rank_level1: [1, -2, -3],
+          rank_level2: [-4, 2, -1],
+          rank_level3: [-3, -1, 5]
+        }, index: ['I', 'II', 'III'])
+      end
+      let(:dv) { Daru::Vector.new ['III', 'II', 'I', 'II', 'II'],
+        name: :rank, type: :category }
+      subject { dv.contrast_code user_defined: df }
+
+      it { is_expected.to be_a Daru::DataFrame }
+      its(:shape) { is_expected.to eq [5, 3] }
+      its(:'rank_level1.to_a') { is_expected.to eq [-3, -2, 1, -2, -2] }
+      its(:'rank_level2.to_a') { is_expected.to eq [-1, 2, -4, 2, 2] }
+      its(:'rank_level3.to_a') { is_expected.to eq [5, -1, -3, -1, -1] }
+    end
+
+    context 'naming' do
+      context "string" do
+        let(:dv) { Daru::Vector.new [:a, 1, :a, 1, :c], type: :category, name: 'abc' }
+        subject { dv.contrast_code }
+        
+        it { is_expected.to be_a Daru::DataFrame }
+        its(:'vectors.to_a') { is_expected.to eq ['abc_1', 'abc_c'] }
+      end
+      
+      context "symbol" do
+        let(:dv) { Daru::Vector.new [:a, 1, :a, 1, :c], type: :category, name: :abc }
+        subject { dv.contrast_code }
+        
+        it { is_expected.to be_a Daru::DataFrame }
+        its(:'vectors.to_a') { is_expected.to eq [:abc_1, :abc_c] }    
+      end      
+    end
+  end
+
+  context '#inspect' do
```

Comment:
It is ok to make some folder like `spec/shared/` and store shared examples there.

In fact, I think it would not be much harm in splitting specs in more files/folders :) They are pretty long now.